README.md

```diff
@@ -1,2 +1,19 @@
 # comp110-worksheet-5
-Base repository for COMP110 worksheet 5
+
+a.) Checks for duplicate elements in a list
+
+b.) If there are no duplicate elements, the algorithm has to check the entire list for each element in the list, I.e.: n times by n
+
+c.) The algorithm would simply not check previous values of i, but would still function as intended
+
+d.) The algorithm won't waste time rechecking previous values, so the algorithm is completed nearly twice as fast
+
+e.) The time complexity of the algorithm is still quadratic, because the maximum running time is now (n^2)/2 which is still quadratic
+
+f.) O(nlogn) (https://stackoverflow.com/questions/14434490/what-is-the-complexity-of-this-python-sort-method)
+
+g.) O(n) becasue after the list has been sorted the algorithm only needs to go through the list once to check for any duplicate elements next to each other
```

Comment:
However the time taken to sort the list needs to be factored in, so the overall time complexity is `O(n log n + n) = O(n log n)`